request-management-api/request_api/services/documentservice.py

```diff
@@ -51,6 +52,23 @@ def createrequestdocumentversion(self, requestid, documentid, documentschema, us
         else:
             return self.createrawdocumentversion(requestid, documentid, documentschema, userid)
 
+    def copyrequestdocumenttonewlocation(self, newcategory, documentpath): #documentpath is full url including https://
+        # for old documentpath
+        baseurl = 'https://' + os.getenv("OSS_S3_HOST")
+        location = documentpath.split('/')[3:] # bucket and filename
+        bucket = location[0]
+        source = "/".join(location) # /bucket/filename
+        # for new document path, replace category name in path
+        newlocation = location
+        newlocation[3] = newcategory
+        filename = "/".join(newlocation[1:])
+        newdocumentpath = "/".join(baseurl) + '/' + bucket + '/' + filename
+        moveresponse = storageservice().copy_file(source, bucket, filename)
+        if moveresponse['ResponseMetadata']['HTTPStatusCode'] == 200:
+            return {"status": "success", 'documentpath': newdocumentpath}
+        else:
+            return {"status": "error"}
+
```

Comment:
It looks like we can't use raise_for_status() here, it's part of the Python requests module, that's not being used to make the S3 request, we're using the boto3 client instead.

Reply:
for ref:
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/copy_object.html